File: db4oj/core/src/com/db4o/internal/fieldindex/OrIndexedLeaf.java
/* Copyright (C) 2006   db4objects Inc.   http://www.db4o.com */

package com.db4o.internal.fieldindex;

import com.db4o.internal.query.processor.*;

public class OrIndexedLeaf extends JoinedLeaf {
	
	public OrIndexedLeaf(QCon constraint, IndexedNodeWithRange leaf1, IndexedNodeWithRange leaf2) {
		super(constraint, leaf1, leaf1.getRange().union(leaf2.getRange()));
	}
}

File: db4on/core/src/core/com/db4o/internal/fieldindex/OrIndexedLeaf.cs
namespace com.db4o.@internal.fieldindex
{
	public class OrIndexedLeaf : com.db4o.@internal.fieldindex.JoinedLeaf
	{
		public OrIndexedLeaf(com.db4o.@internal.query.processor.QCon constraint, com.db4o.@internal.fieldindex.IndexedNodeWithRange
			 leaf1, com.db4o.@internal.fieldindex.IndexedNodeWithRange leaf2) : base(constraint
			, leaf1, leaf1.GetRange().Union(leaf2.GetRange()))
		{
		}
	}
}
